Add Blog Post for KEP-3202 beta (CVE feed) (#39644)
kubernetes/website: content/en/blog/_posts/2023-03-08-Updates-to-the-Auto-refreshing-Official-CVE-Feed/index.md

---

> layout: blog 

Reviewer: ```suggestion
layout: blog
```
nit remove space

title: Updates to the Auto-refreshing Official CVE Feed

> date: 2023-02-08

Reviewer: Reminder that we will have to update this on publication.

slug: k8s-cve-feed-beta
---

> **Author**: Cailyn Edwards (Shopify)

Reviewer: ```suggestion
**Authors**: Cailyn Edwards (Shopify), Pushkar Joglekar (Credit Karma), Mahé Tardy (Isovalent)
```
Is @PushkarJ ok with displaying his employer?

Since launching the [Auto-refreshing Official CVE feed](/docs/reference/issues-security/official-cve-feed/) as an `alpha`

> feature in the 1.25 release we have made signficant improvments and updates. We are excited to announce the release of the 
> `beta` version of the feed. This blog post will outline the changes made, and talk about what is planned for the to expect for 
> the `stable` release.

Reviewer: ```suggestion
feature in the 1.25 release, we have made significant improvements and updates. We are excited to announce the release of the
`beta` version of the feed. This blog post will outline the feedback received, the changes made, and talk about what is
planned for the `stable` release.
```

## Updates
| **\#** | **Title**                                                                                                    | **Issue**                                                       | **Status**                                                                                                                                                                                                                      |
| ------ | ------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| 1 | Support RSS feeds by generating data in Atom format | [kubernetes/sig-security#77](https://github.com/kubernetes/sig-security/issues/77) | open, addressed by [kubernetes/website#39513](https://github.com/kubernetes/website/pull/39513)|

> | 2 | CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | open, no PR open |

Reviewer 2: This PR fixes it: https://github.com/kubernetes/sig-security/pull/76

Reviewer: Did it? I did not realize 😅

| 3 | CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | open, no PR open |
| 4 | CVE Feed: Add lastUpdatedAt as a metadata field | [kubernetes/sig-security#72](https://github.com/kubernetes/sig-security/issues/72) | open, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| 5 | CVE Feed: JSON feed should pass jsonfeed spec validator | [kubernetes/webite#36808](https://github.com/kubernetes/website/issues/36808) | open, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |

> | 6 | CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | open, no PR |

Reviewer 2: Covered by https://github.com/kubernetes/sig-security/pull/76

> ## Updates
> | **\#** | **Title**                                                                                                    | **Issue**                                                       | **Status**                                                                                                                                                                                                                      |
> | ------ | ------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
> | 1 | Support RSS feeds by generating data in Atom format | [kubernetes/sig-security#77](https://github.com/kubernetes/sig-security/issues/77) | closed, addressed by [kubernetes/website#39513](https://github.com/kubernetes/website/pull/39513)|
> | 2 | CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
> | 3 | CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | open, no PR open |
> | 4 | CVE Feed: Add lastUpdatedAt as a metadata field | [kubernetes/sig-security#72](https://github.com/kubernetes/sig-security/issues/72) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
> | 5 | CVE Feed: JSON feed should pass jsonfeed spec validator | [kubernetes/webite#36808](https://github.com/kubernetes/website/issues/36808) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
> | 6 | CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | open, no PR |

Reviewer: ```suggestion
```
I would move that table to later in the article

> 
> ## Summary of Changes
> TODO - add details of changes

Reviewer: ```suggestion
## Feedback from end-users

SIG Security received some feedback from end-users:
- The JSON CVE Feed [did not comply](https://github.com/kubernetes/website/issues/36808)
  with the [JSON Feed specification](https://www.jsonfeed.org/) as its name
  would suggest.
- The feed could also [support RSS](https://github.com/kubernetes/sig-security/issues/77)
  in addition to JSON Feed format.
- Some metadata fields could be added to indicate the freshness of
  [the whole feed](https://github.com/kubernetes/sig-security/issues/72), or
  [specific CVEs](https://github.com/kubernetes/sig-security/issues/63). Also
  indicating [the Prow job](https://github.com/kubernetes/sig-security/issues/71)
  that recently updated the feed. See more ideas directly on the
  [the umbrella issue](https://github.com/kubernetes/sig-security/issues/1).
- The feed markdown table on the website [should be ordered](https://github.com/kubernetes/sig-security/issues/73)
  from the most recent to the least recently announced CVE.

## Summary of changes

In response, the SIG [did a rework of the script generating the JSON feed](https://github.com/kubernetes/sig-security/pull/76)
to comply with the JSON Feed specification from generation and add a
`last_updated` root field to indicate overall freshness. This redesign needed a
[corresponding fix on the Kubernetes website side](https://github.com/kubernetes/website/pull/38579)
for the CVE feed page to continue to work with the new format.

After that, [RSS feed support](https://github.com/kubernetes/website/pull/39513)
could be added transparently so that end-users can consume the feed in their
preferred format.

Overall, the redesign based on the JSON Feed specification, which this time broke
backward compatibility, will allow updates in the future to address the rest of
the issue while being more transparent and less disruptive to end-users.
```

> 

Reviewer: ```suggestion

### Updates
| **Title**                                                                                                    | **Issue**                                                       | **Status**                                                                                                                                                                                                                      |
| ------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| CVE Feed: JSON feed should pass jsonfeed spec validator | [kubernetes/webite#36808](https://github.com/kubernetes/website/issues/36808) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| CVE Feed: Add lastUpdatedAt as a metadata field | [kubernetes/sig-security#72](https://github.com/kubernetes/sig-security/issues/72) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| Support RSS feeds by generating data in Atom format | [kubernetes/sig-security#77](https://github.com/kubernetes/sig-security/issues/77) | closed, addressed by [kubernetes/website#39513](https://github.com/kubernetes/website/pull/39513)|
| CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | open, no PR |
| CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | open, no PR open |

```

I would reorder the table and remove the number since it's not really ordered (the first column).

> ## What's Next?

Reviewer: ```suggestion
## What's next?
```

> | 7 | CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | open, no PR |

Reviewer 2: This is repeat of `2` unless I missed it

> TODO - add details of changes

Reviewer 2: @cailynse would it be helpful if I or @mtardy can add a `suggestion` that detail the changes here which you can commit directly into the PR?

In preparation to graduate this feature, SIG Security
is still gathering feedback from end users who are using the updated beta feed.

To help us continue to improve the feed in future Kubernetes Releases please share feedback by adding a comment to
this [tracking issue](https://github.com/kubernetes/sig-security/issues/1) or
let us know on
[#sig-security-tooling](https://kubernetes.slack.com/archives/C01CUSVMHPY)

> Kubernetes Slack channel.
> (Join [Kubernetes Slack here](https://slack.k8s.io))

Reviewer: ```suggestion
Kubernetes Slack channel, join [Kubernetes Slack here](https://slack.k8s.io).
```

---
layout: blog
title: Updates to the Auto-refreshing Official CVE Feed

> date: 2023-03-22

Reviewer 2: Typically, #sig-docs-blog folks + release team, will assign a date for this closer to or soon after the release is GA, let's make sure to change to that here and in the markdown file name once we know that date.

Reviewer: btw I asked there https://kubernetes.slack.com/archives/CNT9Y603D/p1679485223404749

slug: k8s-cve-feed-beta
---

> **Authors**: Cailyn Edwards (Shopify), Pushkar Joglekar (Credit Karma), Mahé Tardy (Isovalent)

Reviewer 2: ```suggestion
**Authors**: Cailyn Edwards (Shopify), Mahé Tardy (Isovalent), Pushkar Joglekar
```

I am working in personal capacity so do not want to have employer name here.

Reviewer 2: Switched the sequence to make it _alphabetical_

Reviewer: ah! I made it alphabetical on the last name

> Since launching the [Auto-refreshing Official CVE feed](/docs/reference/issues-security/official-cve-feed/) as an alpha

Reviewer 2: ```suggestion
Since launching the [Auto-refreshing Official CVE feed](/docs/reference/issues-security/official-cve-feed/) as an `alpha`
```

> feature in the 1.25 release, we have made significant improvements and updates. We are excited to announce the release of the
> beta version of the feed. This blog post will outline the feedback received, the changes made, and talk about what is
> planned for the stable release.

Reviewer 2: ```suggestion
feature in the 1.25 release, we have made significant improvements and updates. We are excited to announce the release of the
`beta` version of the feed. This blog post will outline the feedback received, the changes made, and talk about how you can help as we prepare to make this a `stable` feature in a future Kubernetes Release
```

Author: Will wait for a decision on the `highlight` and then address

  the feed overall, or [specific CVEs](https://github.com/kubernetes/sig-security/issues/63). Another suggestion was 
  to [indicate](https://github.com/kubernetes/sig-security/issues/71) which Prow job recently updated the feed. See 
  more ideas directly on the [the umbrella issue](https://github.com/kubernetes/sig-security/issues/1).
- The feed Markdown table on the website [should be ordered](https://github.com/kubernetes/sig-security/issues/73)
  from the most recent to the least recently announced CVE.

## Summary of changes

In response, the SIG did a [rework of the script generating the JSON feed](https://github.com/kubernetes/sig-security/pull/76)
to comply with the JSON Feed specification from generation and add a
`last_updated` root field to indicate overall freshness. This redesign needed a
[corresponding fix on the Kubernetes website side](https://github.com/kubernetes/website/pull/38579)
for the CVE feed page to continue to work with the new format.

After that, [RSS feed support](https://github.com/kubernetes/website/pull/39513)
could be added transparently so that end-users can consume the feed in their
preferred format.

> Overall, the redesign based on the JSON Feed specification, which this time broke

Reviewer 2: Thank you, this is a great callout!

| CVE Feed: JSON feed should pass jsonfeed spec validator | [kubernetes/webite#36808](https://github.com/kubernetes/website/issues/36808) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| CVE Feed: Add lastUpdatedAt as a metadata field | [kubernetes/sig-security#72](https://github.com/kubernetes/sig-security/issues/72) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| Support RSS feeds by generating data in Atom format | [kubernetes/sig-security#77](https://github.com/kubernetes/sig-security/issues/77) | closed, addressed by [kubernetes/website#39513](https://github.com/kubernetes/website/pull/39513)|
| CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |

> | CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | open, no PR |

Reviewer 2: This was fixed by addition of `date_published` 

```suggestion
| CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76)  |
```

> | CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | open, no PR open |

Reviewer 2: ```suggestion
| CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | closed, addressed by [kubernetes/sig-security#83](https://github.com/kubernetes/sig-security/pull/83) |
```

## What's next?

> In preparation to graduate the feed to general availability, SIG Security

Reviewer 2: ```suggestion
In preparation to [graduate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages) the feed to _stable_ i.e. `General Availability` stage, SIG Security
```

> is still gathering feedback from end users who are using the updated beta feed.

Reviewer 2: ```suggestion
is still gathering feedback from end users who are using the updated `Beta` feed.
```
_highlighted as code_ because it refers to a specific feature stage

Reviewer 2: I will trust @sftim on this, so feel free to ignore this type of other suggestions from me.

---
layout: blog
title: Updates to the Auto-refreshing Official CVE Feed

> date: 2023-03-17

Reviewer 3: ```suggestion
date: 2023-03-22
```

There's a big [change](https://kubernetes.io/blog/2023/03/10/image-registry-redirect/) on Monday. Let's drop this one in after that's happened.

slug: k8s-cve-feed-beta
---

**Authors**: Cailyn Edwards (Shopify), Pushkar Joglekar (Credit Karma), Mahé Tardy (Isovalent)

Since launching the [Auto-refreshing Official CVE feed](/docs/reference/issues-security/official-cve-feed/) as an `alpha`
feature in the 1.25 release, we have made significant improvements and updates. We are excited to announce the release of the
`beta` version of the feed. This blog post will outline the feedback received, the changes made, and talk about what is

> planned for the `stable` release.

Reviewer 3: nit: don't write `alpha` etc in backticks.

Author: Same for `beta` and `stable`?

## Feedback from end-users

SIG Security received some feedback from end-users:
- The JSON CVE Feed [did not comply](https://github.com/kubernetes/website/issues/36808)
  with the [JSON Feed specification](https://www.jsonfeed.org/) as its name
  would suggest.
- The feed could also [support RSS](https://github.com/kubernetes/sig-security/issues/77)
  in addition to JSON Feed format.

> - Some metadata fields could be added to indicate the freshness of
>   [the whole feed](https://github.com/kubernetes/sig-security/issues/72), or

Reviewer 3: (nit)
```suggestion
- Some metadata could be [added](https://github.com/kubernetes/sig-security/issues/72) to
  indicate the freshness of the feed overall.
```
?

  [specific CVEs](https://github.com/kubernetes/sig-security/issues/63). Also

>   indicating [the Prow job](https://github.com/kubernetes/sig-security/issues/71)

Reviewer 3: (nit)
```suggestion
  [indicating](https://github.com/kubernetes/sig-security/issues/71) which Prow job
```

  that recently updated the feed. See more ideas directly on the
  [the umbrella issue](https://github.com/kubernetes/sig-security/issues/1).

> - The feed markdown table on the website [should be ordered](https://github.com/kubernetes/sig-security/issues/73)

Reviewer 3: (nit)
```suggestion
- The feed Markdown table on the website [should be ordered](https://github.com/kubernetes/sig-security/issues/73)
```

  from the most recent to the least recently announced CVE.

## Summary of changes

> In response, the SIG [did a rework of the script generating the JSON feed](https://github.com/kubernetes/sig-security/pull/76)

Reviewer 3: ```suggestion
In response, the SIG did a [rework of the script generating the JSON feed](https://github.com/kubernetes/sig-security/pull/76)
```

to comply with the JSON Feed specification from generation and add a
`last_updated` root field to indicate overall freshness. This redesign needed a
[corresponding fix on the Kubernetes website side](https://github.com/kubernetes/website/pull/38579)
for the CVE feed page to continue to work with the new format.

After that, [RSS feed support](https://github.com/kubernetes/website/pull/39513)
could be added transparently so that end-users can consume the feed in their
preferred format.

Overall, the redesign based on the JSON Feed specification, which this time broke
backward compatibility, will allow updates in the future to address the rest of
the issue while being more transparent and less disruptive to end-users.

> ### Updates

Reviewer 3: ```suggestion
### Progress since the alpha
```
?

Reviewer: It's also things that are not yet address

| CVE Feed: Add lastUpdatedAt as a metadata field | [kubernetes/sig-security#72](https://github.com/kubernetes/sig-security/issues/72) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| Support RSS feeds by generating data in Atom format | [kubernetes/sig-security#77](https://github.com/kubernetes/sig-security/issues/77) | closed, addressed by [kubernetes/website#39513](https://github.com/kubernetes/website/pull/39513)|
| CVE Feed: Sort Markdown Table from most recent to least recently announced CVE | [kubernetes/sig-security#73](https://github.com/kubernetes/sig-security/issues/73) | closed, addressed by [kubernetes/sig-security#76](https://github.com/kubernetes/sig-security/pull/76) |
| CVE Feed: Include a timestamp field for each CVE indicating when it was last updated | [kubernetes/sig-security#63](https://github.com/kubernetes/sig-security/issues/63) | open, no PR |
| CVE Feed: Add Prow job link as a metadata field | [kubernetes/sig-security#71](https://github.com/kubernetes/sig-security/issues/71) | open, no PR open |

## What's next?

> In preparation to graduate this feature, SIG Security

Reviewer 3: ```suggestion
In preparation to graduate the feed to general availability, SIG Security
```
